src/eosjs2-serialize.ts

```diff
@@ -559,74 +573,122 @@ function deserializeUnknown(buffer: SerialBuffer): SerialBuffer {
     throw new Error("Don't know how to deserialize " + this.name);
 }
 
-function serializeStruct(buffer: SerialBuffer, data: any) {
+function serializeStruct(this: Type, buffer: SerialBuffer, data: any,
```

Comment:
Why the need to add `this` to a lot of these functions?

Reply:
It silently defaulted to `any`, defeating some type checking

Reply:
Oddly, the no-implicit-any check didn't catch it

Reply:
Interesting, I would not have expected this as the solution to that problem. Reminds me of Python's `self`.

Reply:
The weird part is it's still a hidden argument

Reply:
https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters

It's a neat TS feature.